File: mobile/ios/Hearth/Core/Models/AuthResponse.swift
import Foundation

struct AuthResponse: Codable, Sendable {
    let token: String
    let refreshToken: String
    let user: User

    enum CodingKeys: String, CodingKey {
        case token
        case refreshToken = "refresh_token"
        case user
    }
}

struct LoginRequest: Codable, Sendable {
    let email: String
    let password: String
}

struct RegisterRequest: Codable, Sendable {
    let email: String
    let username: String
    let password: String
}

struct RefreshTokenRequest: Codable, Sendable {
    let refreshToken: String

    enum CodingKeys: String, CodingKey {
        case refreshToken = "refresh_token"
    }
}

struct SendMessageRequest: Codable, Sendable {
    let content: String
}

struct CreateDMRequest: Codable, Sendable {
    let recipientId: String

    enum CodingKeys: String, CodingKey {
        case recipientId = "recipient_id"
    }
}

File: mobile/android/app/src/main/java/co/hndrx/hearth/core/models/AuthResponse.kt
package co.hndrx.hearth.core.models

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class AuthResponse(
    @Json(name = "token") val token: String,
    @Json(name = "refresh_token") val refreshToken: String,
    @Json(name = "user") val user: User,
)

@JsonClass(generateAdapter = true)
data class LoginRequest(
    @Json(name = "email") val email: String,
    @Json(name = "password") val password: String,
    @Json(name = "mfa_code") val mfaCode: String? = null,
)

@JsonClass(generateAdapter = true)
data class RegisterRequest(
    @Json(name = "email") val email: String,
    @Json(name = "username") val username: String,
    @Json(name = "password") val password: String,
)

@JsonClass(generateAdapter = true)
data class RefreshTokenRequest(
    @Json(name = "refresh_token") val refreshToken: String,
)

@JsonClass(generateAdapter = true)
data class SendMessageRequest(
    @Json(name = "content") val content: String,
    @Json(name = "reply_to") val replyTo: String? = null,
    @Json(name = "tts") val tts: Boolean = false,
)

@JsonClass(generateAdapter = true)
data class CreateDMRequest(
    @Json(name = "recipient_id") val recipientId: String,
)
